tools/ci/precheck.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+set -x
+
+PRECHECK_URL="https://ci.inria.fr/ocaml/job/precheck"
+
+# A filename used to pipe data between commands
+OUTPUT="out.txt"
+
+# Bail out if comment is not trigger
+
+if test "${COMMENT_BODY}" != "/precheck"; then exit 0; fi
+
+# Bail out if not in a PR
+
+if test -z "${PULL_REQUEST_URL}"; then exit 0; fi
+
+# Check that the user belongs to caml-devel
+
+## Retrieve team URL
+
+curl -s \
+     -H "Authorization: Bearer ${CAML_DEVEL_TOKEN}" \
+     -H "Content-type: application/json" \
+     https://api.github.com/orgs/ocaml/teams/caml-devel | jq -r '.url' | tee ${OUTPUT}
+
+read CAML_DEVEL_URL < ${OUTPUT}
+
+echo "CAML_DEVEL_URL=${CAML_DEVEL_URL}"
+echo "COMMENT_USER_LOGIN=${COMMENT_USER_LOGIN}"
+
+## Retrieve the list of members of caml-devel
+
+curl -s \
+     -H "Authorization: Bearer ${CAML_DEVEL_TOKEN}" \
+     -H "Content-type: application/json" \
+     ${CAML_DEVEL_URL}/members | jq -r '.[].login' > ${OUTPUT}
+
+## Check whether comment author belongs to caml-devel
+
+if ! grep -Fxq "${COMMENT_USER_LOGIN}" ${OUTPUT}
+then
+    MESSAGE="Sorry, only members of the developer team may trigger precheck."
+    echo "$MESSAGE" | jq -R -s '{body: .}' | \
+        curl -s -H "Authorization: Bearer ${GITHUB_TOKEN}" \
+             -H "Content-type: application/json" \
+             ${ISSUE_URL}/comments -d @-
+    exit 0
+fi
```

Comment:
All this code makes me feel like listing the authorized users straight inside the script!

Reply:
Good point! Will amend.